### PR TITLE
Deploy to AWS

### PR DESCRIPTION
**Pull request by brolewis**

### CODE REVIEW

File: src/api/data.py [hunk 1]
```diff
@@ -132,6 +130,7 @@ def get_aggregate(self):
             cursor.execute(statement)
             aggregate_data["columns"] = [x.name for x in cursor.description]
             aggregate_data["data"] = cursor.fetchall()
+        self.connection.close()
```
Comment (brolewis) [inline src/api/data.py:133]: Turns out you have to explicitly call `close()` to actually release the connection and not releasing was causing problems when trying to delete the database when deleting a stack. That's also why all the methods were set to private to make clear where the `close()` is being called.

File: src/api/aws.py [hunk 2]
```diff
@@ -32,6 +33,7 @@ def get_database_url() -> str:
     }
     if db_info["engine"] == "postgres":
         db_info["engine"] = "postgresql"
+    db_info["password"] = urllib.parse.quote(db_info["password"])
```
Comment (bradsbrown) [inline src/api/aws.py:36]: 💯

File: scripts/api_template.yml [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+  ApiDbManagementLambda:
+    Type: AWS::Serverless::Function
+    Properties:
+      CodeUri: ../src/.
+      Description: DC Aggregation Database Management Lambda
+      Environment:
+        Variables:
+          DATABASE_NAME: !Sub "${DatabaseName}"
+          DB_SECRET_NAME:
+            Fn::ImportValue: "sdn-shared-stack:ExportsOutputRefDbSecret685A0FA57881E1C6"
```
Comment (bradsbrown) [inline scripts/api_template.yml:90]: One suggestion for a future tweak -- and feel free to hate it -- but all of these values imported from `sdn-shared-stack` with the long names here feel a bit hinky in the resource definitions. I'd suggest considering moving the value names themselves (i.e. `ExportsOutputRefDbSecret685A0FA57881E1C6`) into a [mapping](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/mappings-section-structure.html) at the top of the file, and just doing a `!FindInMap` to pull the name in here.

**Review by bradsbrown — APPROVED** [review]
LGTM

**Pull request merged**